pkgs/applications/misc/squeekboard/default.nix

```diff
@@ -0,0 +1,84 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, cmake
+, meson
+, ninja
+, pkgconfig
+, gnome3
+, glib
+, gtk3
+, wayland
+, wayland-protocols
+, rustc
+, cargo
+, libcroco
+, libxml2
+, libxkbcommon
+, rustPlatform
+, makeWrapper
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "squeekboard";
+  version = "1.9.1";
+
+  src = fetchFromGitLab {
+    domain = "source.puri.sm";
+    owner = "Librem5";
+    repo = pname;
+    rev = "334898c5a5d45a5883654b4f2f07f28ac8f2bfc4";
+    sha256 = "1nrihpffr91p3ipif5k5z9vcvkwxk6qy8hp93w13s7gnwl2g93hx";
+  };
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkgconfig
+    rustc
+    cargo
+    glib.dev
+    gtk3.dev
+    wayland
+    gnome3.gnome-desktop
+    makeWrapper
+  ];
+
+  buildInputs = [
+    glib.dev
+    gtk3.dev
+    gnome3.gnome-desktop
+    wayland
+    wayland-protocols
+    libcroco
+    libxml2
+    libxkbcommon
+  ];
+
+  cargoSha256 = "1fkhj4i2l2hdk9wvld6ryvnm1mxfwx3s555r7n42pg9f5namn1sr";
+
+  NIX_CFLAGS_COMPILE = "-I${glib.dev}/include/gio-unix-2.0";
+
+  configurePhase = "meson . _build/";
```

Comment:
This probably needs some explanation on how this is set up. I saw some references to `_build` in the upstream `.gitlab-ci.yml` - I assume meson can't build all of this?

Reply:
This is sometimes needed when the project uses `build` directory for auxiliary build scripts (not a good idea since `build` directory is conventionally used as Meson out-of-tree build directory, `build-aux` should be used instead). We should really make it directory name configurable in https://github.com/NixOS/nixpkgs/blob/fffa6e8c1872d8e6f7dfb94577b871a2952505d4/pkgs/development/tools/build-managers/meson/setup-hook.sh#L23-L24

But here, I do not see `build` directory in the repo, so there should not be a conflict.

Reply:
I tried building the derivation without using custom `configurePhase`, `buildPhase` and `installPhase` and it builts, but the output bin is `test_layout` and I can't find where is the `squeekboard` binary. Any idea?

Reply:
Does this differ from the previous state?

Reply:
I mean, if I build using the custom phases I obtain the `squeekboard` binary and it works fine. But If I remove the three custom phases, the resulting derivation only contains a `bin/test_layout` and `lib/librs.a`.

Reply:
Oh, did not notice this is `buildRustPackage`. In that case you need to set the phases to `null` to use the ones from `stdenv.mkDerivation`, see e.g. fractal expression:

https://github.com/NixOS/nixpkgs/blob/612d330c67e3b24625c2707edd0b29dfe08a8483/pkgs/applications/networking/instant-messengers/fractal/default.nix#L75-L79